firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleListener.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.bundle;
+
+import com.google.firebase.database.collection.ImmutableSortedMap;
+import com.google.firebase.database.collection.ImmutableSortedSet;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.MaybeDocument;
+
+/** Interface implemented by components that can apply changes from a bundle. */
+public interface BundleListener {
```

Comment:
Maybe call it `BundleStorage`? And update comment to `...that can apply data from bundles to local storage`?

Reply:
We seem to call these types of interfaces "Callback": OnlineStateCallback, SyncEngineCallback, StreamCallback, RemoteStoreCallback. I have updated this to use "callback" as well, but also used your comment enhancement as it still applies.